### DISCUSSION TITLE
Pytest fixtures x standard Python objects

### DISCUSSION THREAD

**Discussion by staticdev** [pycabook/pycabook#35] [Ideas]
Hi @lgiordani!

I am re-reading your book in the second edition draft and noticed a change in some tests to use normal Python variables instead of pytest fixtures. Eg.

tests/test_responses.py (2st edition):

`SUCCESS_VALUE = {"key": ["value1", "value2"]}`

tests/test_response_objects.py (1st edition):

```
@pytest.fixture
def response_value():
    return {"key": ["value1", "value2"]}
```

It makes the code cleaner and maybe (just want to test) test execution faster.

Still there are some pytest fixtures like in chapter 5.

tests/use_cases/test_room_list.py:

```
def domain_rooms():
    room_1 = r.Room(
        code=uuid.uuid4(),
        size=215,
        price=39,
        longitude=-0.09998975,
        latitude=51.75436293,
    )

    room_2 = r.Room(
        code=uuid.uuid4(),
        size=405,
        price=66,
        longitude=0.18228006,
        latitude=51.74640997,
    )

    room_3 = r.Room(
        code=uuid.uuid4(),
        size=56,
        price=60,
        longitude=0.27891577,
        latitude=51.45994069,
    )

    room_4 = r.Room(
        code=uuid.uuid4(),
        size=93,
        price=48,
        longitude=0.33894476,
        latitude=51.39916678,
    )

    return [room_1, room_2, room_3, room_4]
```

Just curious, do you intent do replace them all? Is it going to be another branch for https://github.com/pycabook/rentomatic? I may be able to contribute is you are looking for this kind collaboration.

Kind regards,

Thiago

**Comment by lgiordani** [comment]
Hi Thiago, yes I'm replacing some of the fixtures because what I realised in these last years is that in general fixtures should be values that need to be reset at the beginning of tests because you plan to potentially change them. The work on the second edition is still going on, though, so I can't say what is going to happen to other fixtures. I will definitely create a new branch on rentomatic, I don't want to make clones that people already have inconsistent.
Thanks for the offer. On the book I prefer to keep my pace and see where I can get without pressure. I'm already falling back to follow what people send me for the blog and other projects :facepalm:

**Comment by staticdev** [comment]
I understand, I am also working on many projects at the same time. No shame. May the force be with you!

**Comment by lgiordani** [comment]
Many thanks for the suggestions anyway, I very much appreciate the time you take to help me to improve the book!
